utils/hostname.py

```diff
@@ -72,6 +72,12 @@ def get_hostname(config=None):
         if is_valid_hostname(gce_hostname):
             return gce_hostname
 
+    # Try to get Azure VM ID
+    azure_hostname = Azure.get_hostname(config)
+    if azure_hostname is not None:
+        if is_valid_hostname(azure_hostname):
+            return azure_hostname
```

Comment:
Actually we don't want to change the canonical hostname that's used by the agent (as this could cause hosts to change names in the app, and would require a complex migration process on our part and our users' part).

Instead of using the azure VM ID as the hostname, we should only add it to the host aliases in the metadata payload here: https://github.com/DataDog/dd-agent/blob/5.14.1/checks/collector.py#L757-L759

Let me know if you agree/disagree or have any questions :)